pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -901,6 +903,35 @@ public String getServerMetadata(
     return segmentsMetadata;
   }
 
+  @GET
+  @Path("segments/{tableName}/zkmetadata")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Get the zookeeper metadata for all table segments", notes = "Get the zookeeper metadata for "
+      + "all table segments")
+  public String getZookeeperMetadata(
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr)
+      throws JsonProcessingException {
+    LOGGER.info("Received a request to fetch zookeeper metadata for all segments for table {}", tableName);
+    TableType tableType = Constants.validateTableType(tableTypeStr);
+
+    List<String> tableNamesWithType =
+        ResourceUtils.getExistingTableNamesWithType(_pinotHelixResourceManager, tableName, tableType, LOGGER);
+    Map<String, JsonNode> tableTypeToSegmentMetadata = new HashMap<>();
+    for (String tableNameWithType : tableNamesWithType) {
+      List<SegmentZKMetadata> segmentZKMetadataList =
+          _pinotHelixResourceManager.getSegmentsZKMetadata(tableNameWithType);
+      Map<String, JsonNode> segmentToMetadataMap = new HashMap<>();
+      for (SegmentZKMetadata segmentZKMetadata : segmentZKMetadataList) {
+        segmentToMetadataMap.put(segmentZKMetadata.getSegmentName(), JsonUtils.objectToJsonNode(segmentZKMetadata));
```

Comment:
`SegmentZKMetadata` is not designed to be JSON serializable. We can use `toMap() ` extract the values.
```suggestion
        segmentToMetadataMap.put(segmentZKMetadata.getSegmentName(), segmentZKMetadata.toMap());
```